ovos_PHAL_plugin_sj201_led/__init__.py

```diff
@@ -87,11 +91,23 @@ def turn_off(self):
 class MycroftSJ201Validator:
     @staticmethod
     def validate(config=None):
-        """ this method is called before loading the plugin.
-        If it returns False the plugin is not loaded.
-        This allows a plugin to run platform checks"""
-        # TODO detect sj201
-        return True
+        # If the user enabled the plugin no need to go further
+        if config.get("enabled"):
+            return True
+        # Check for a file created by ovos-i2csound
+        # https://github.com/OpenVoiceOS/ovos-i2csound/blob/a107b74d61bbb362421fb51da6b91f2ba49b6a1a/ovos-i2csound#L76
+        LOG.debug(f"checking file {I2C_PLATFORM_FILE}")
+        if exists(I2C_PLATFORM_FILE):
+            with open(I2C_PLATFORM_FILE) as f:
+                platform = f.readline().strip().lower()
+            if platform == "SJ201LED":
+                return True
+        # Try a direct hardware check
+        if is_mycroft_sj201():
```

Comment:
has this been validated? not sure this plugin works at all, and if it does i think ONLY in devkit, so we need to exclude rev10 boards

Reply:
This ONLY allows the dev kit.  I just added validation, have not tested the plugin
reference:
https://github.com/OpenVoiceOS/ovos-PHAL/blob/09f8b33460d2c785623f8b1c0ab73d50762064da/ovos_PHAL/detection.py#L5

But just noticed i2csound needs fixed for each board version.

Reply:
ovos-i2csound was fixed [here](https://github.com/OpenVoiceOS/ovos-i2csound/commit/03259b5d4e4521d0189df0d529c0af87be20697b)

With this PR, any auto-detection that I can think of is handled.